src/image/image.c

```diff
@@ -171,10 +171,140 @@ k4a_result_t image_create(k4a_image_format_t format,
+        }
+        else if (stride_bytes < 1 * width_pixels)
+        {
+            LOG_ERROR("Insufficient stride (%d bytes) to represent image width (%d pixels).",
+                      stride_bytes,
+                      width_pixels);
+            result = K4A_RESULT_FAILED;
+        }
+        else
+        {
+            // Calculate correct size for NV12 (extra color lines follow Y samples)
+            size = 3 * (size_t)height_pixels * (size_t)stride_bytes / 2;
+            result = K4A_RESULT_SUCCEEDED;
+        }
+        break;
+    }
+
+    // 1 Byte per pixel
+    case K4A_IMAGE_FORMAT_CUSTOM8:
+    {
+        if (stride_bytes < 1 * width_pixels)
+        {
+            LOG_ERROR("Insufficient stride (%d bytes) to represent image width (%d pixels).",
+                      stride_bytes,
+                      width_pixels);
+            result = K4A_RESULT_FAILED;
+        }
+        else
+        {
+            size = (size_t)height_pixels * (size_t)stride_bytes;
+            result = K4A_RESULT_SUCCEEDED;
+        }
+        break;
+    }
+
+    // 2 Bytes per pixel
+    case K4A_IMAGE_FORMAT_DEPTH16:
+    case K4A_IMAGE_FORMAT_IR16:
+    case K4A_IMAGE_FORMAT_CUSTOM16:
+    {
+        if (stride_bytes < 2 * width_pixels)
+        {
+            LOG_ERROR("Insufficient stride (%d bytes) to represent image width (%d pixels).",
+                      stride_bytes,
+                      width_pixels);
+            result = K4A_RESULT_FAILED;
+        }
+        else
+        {
+            size = (size_t)height_pixels * (size_t)stride_bytes;
+            result = K4A_RESULT_SUCCEEDED;
+        }
+        break;
+    }
+
+    // 2 Bytes per pixel
+    case K4A_IMAGE_FORMAT_COLOR_YUY2:
+    {
+        if (width_pixels % 2 != 0)
+        {
+            LOG_ERROR("YUY2 requires an even number of pixels per line. Width of %d is invalid.", width_pixels);
+            result = K4A_RESULT_FAILED;
+        }
+        else if (stride_bytes < 2 * width_pixels)
+        {
+            LOG_ERROR("Insufficient stride (%d bytes) to represent image width (%d pixels).",
+                      stride_bytes,
+                      width_pixels);
+            result = K4A_RESULT_FAILED;
+        }
+        else
+        {
+            size = (size_t)height_pixels * (size_t)stride_bytes;
+            result = K4A_RESULT_SUCCEEDED;
+        }
+        break;
+    }
+
+    // 4 Bytes per pixel
+    case K4A_IMAGE_FORMAT_COLOR_BGRA32:
+    {
+        if (stride_bytes < 4 * width_pixels)
+        {
+            LOG_ERROR("Insufficient stride (%d bytes) to represent image width (%d pixels).",
+                      stride_bytes,
+                      width_pixels);
+            result = K4A_RESULT_FAILED;
+        }
+        else
+        {
+            size = (size_t)height_pixels * (size_t)stride_bytes;
+            result = K4A_RESULT_SUCCEEDED;
+        }
+        break;
+    }
+
+    // Unknown
+    case K4A_IMAGE_FORMAT_CUSTOM:
+    default:
```

Comment:
Perhaps we should not support this type. I suppose we already have code though relying on this